src/Microsoft.Health.Fhir.Liquid.Converter.UnitTests/Utilities/ComplexObjectFilterUtilityTests.cs

```diff
@@ -0,0 +1,69 @@
+﻿// -------------------------------------------------------------------------------------------------
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License (MIT). See LICENSE in the repo root for license information.
+// -------------------------------------------------------------------------------------------------
+
+using System.Collections.Generic;
+using System.IO;
+using Microsoft.Health.Fhir.Liquid.Converter.Extensions;
+using Microsoft.Health.Fhir.Liquid.Converter.Utilities;
+using Newtonsoft.Json.Linq;
+using Xunit;
+
+using TestCase = Microsoft.Health.Fhir.Liquid.Converter.UnitTests.Models.ComplexObjectFilterUtilityTestCase;
+
+namespace Microsoft.Health.Fhir.Liquid.Converter.UnitTests.Utilities
+{
+    public class ComplexObjectFilterUtilityTests
+    {
+        [Fact]
+        public void GivenObjectAndPath_SelectCorrectObjectsFromArray()
+        {
+            var testContent = File.ReadAllText("./TestData/ComplexObject.json");
+            var testData = (object[])JToken.Parse(testContent).ToObject();
+            var tests = new TestCase[]
+            {
+                new TestCase(testData, "age", "27", new object[] { testData[0] }),
```

Comment:
Another way to use test cases in Xunit is to use the Theory attribute in conjunction with InlineData, ClassData or MemberData. Here's a [good article](https://andrewlock.net/creating-parameterised-tests-in-xunit-with-inlinedata-classdata-and-memberdata/) on how to do that. Same could be said for the unit test below.

Reply:
This code base does that for the functional tests, but sticks to this method for the unit tests.  I think this is keeping it consistent.

Reply:
Oh, yea I think the functional tests do say which test case failed.

Reply:
That worked really well for the path test but the other test builds the test cases using data read in from a file. Do you have any advice how to achieve that?